Fix SickRage/sickrage-issues/issues/813

fernandog opened the pull request:
Fix https://github.com/SickRage/sickrage-issues/issues/813

User can choose to PP even if no subtitle is available even if setting "postpone if no subs" is enabled
and it wont postpone if show doesn't have subs enabled

@MGaetan89 updates webapi

**UPDATE:**
known issues:
this filename `[HorribleSubs] Boku dake ga Inai Machi - 07 [720p].mkv` raises an error in subliminal causing:

```
FINDSUBTITLES :: Error occurred when downloading subtitles for: /media/[HorribleSubs] Boku dake ga Inai Machi - 07 [720p].mkv. Error: Insufficient data to process the guess
```

when that happen user should manual PP with skipping subtitle check so it can be PP
But even after PP subliminal will still raise an error while searching for subs.

SickRage2 commented:
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked

- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

miigotu commented:
I recall a time when I said this "dont pp if no subs" would result in a never ending line of patches and workarounds to make it work. I thought "Its done, it works perfect"?

fernandog commented on contrib/nzbToMedia:
```diff
@@ -1 +1 @@
-Subproject commit 3e4861e87c543a4d2debd8997feb16bda3ef5339
```
why this got updated? I didn't change

fernandog commented on contrib/nzbToMedia:
```diff
@@ -1 +1 @@
-Subproject commit 3e4861e87c543a4d2debd8997feb16bda3ef5339
+Subproject commit 1cd64053b0da1d351bb6af392e1ebb904c6b6ce2
```
why this got updated? I didn't change

fernandog commented:
Its not a workaround. Its a setting in manual PP like failed, priority, re-process, etc

miigotu commented:
Except what about automatic pp... Show don't have subs enabled but postpone if no subs...

fernandog commented:
@miigotu
Its not the scope of this PR. This fixes part of the issue.
Will do another PR for this other part

miigotu commented:
You cant change params of existing methods in the webapi.

fernandog commented:
It not a required field. It defaults to True
Not an issue

miigotu commented:
it is an issue, because nzbtomedia works by matching the method signature. add/remove/rename a parameter and it stops working.

fernandog commented:
as I said, it works. nzbtomedia calls the url without the parameter.
It took a while because I dont pay for news server. Use a free one.

```
Wed Feb 10 22:59:48 2016	INFO	nzbToMedia: [22:59:48] [INFO]::MAIN: #########################################################
Wed Feb 10 22:59:48 2016	INFO	nzbToMedia: [22:59:48] [INFO]::MAIN: ## ..::[nzbToMedia.py]::.. ##
Wed Feb 10 22:59:48 2016	INFO	nzbToMedia: [22:59:48] [INFO]::MAIN: #########################################################
Wed Feb 10 22:59:48 2016	INFO	nzbToMedia: [22:59:48] [INFO]::MAIN: Script triggered from NZBGet Version 17.0-testing-r1606M.
Wed Feb 10 22:59:48 2016	INFO	nzbToMedia: [22:59:48] [INFO]::MAIN: Auto-detected SECTION:SickBeard
Wed Feb 10 22:59:48 2016	INFO	nzbToMedia: [22:59:48] [INFO]::MAIN: Calling SickBeard:series to post-process:Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated
Wed Feb 10 22:59:48 2016	INFO	nzbToMedia: [22:59:48] [INFO]::MAIN: Attempting to auto-detect series fork
Wed Feb 10 22:59:49 2016	INFO	nzbToMedia: [22:59:49] [INFO]::MAIN: SickBeard:series fork auto-detection successful ...
Wed Feb 10 22:59:49 2016	INFO	nzbToMedia: [22:59:49] [INFO]::MAIN: SickBeard:series fork set to sickrage
Wed Feb 10 22:59:49 2016	INFO	nzbToMedia: [22:59:49] [INFO]::MAIN: FLATTEN: Flattening directory: /media/SAMSUNG/media/downloaded/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated
Wed Feb 10 22:59:51 2016	INFO	nzbToMedia: [22:59:51] [INFO]::TRANSCODER: Checking [Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated.mkv] for corruption, please stand by ...
Wed Feb 10 22:59:52 2016	INFO	nzbToMedia: [22:59:52] [INFO]::TRANSCODER: SUCCESS: [Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated.mkv] has no corruption.
Wed Feb 10 22:59:52 2016	INFO	nzbToMedia: [22:59:52] [POSTPROCESS]::SICKBEARD: SUCCESS: The download succeeded, sending a post-process request
Wed Feb 10 23:00:16 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: Processing folder /media/SAMSUNG/media/downloaded/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated
Wed Feb 10 23:00:16 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: Feature 'postpone postprocessing if no subtitle available' is enabled
Wed Feb 10 23:00:16 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: PostProcessing Path: /media/SAMSUNG/media/downloaded
Wed Feb 10 23:00:16 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: PostProcessing Dirs: [u'Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated']
Wed Feb 10 23:00:16 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: PostProcessing Files: []
Wed Feb 10 23:00:16 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: PostProcessing VideoFiles: []
Wed Feb 10 23:00:16 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: PostProcessing RarContent:
Wed Feb 10 23:00:16 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: PostProcessing VideoInRar:
Wed Feb 10 23:00:16 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: Processing folder Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated
Wed Feb 10 23:00:16 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: Processing folder /media/SAMSUNG/media/downloaded/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated
Wed Feb 10 23:00:16 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: Processing /media/SAMSUNG/media/downloaded/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated.mkv (Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated)
Wed Feb 10 23:00:16 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: Retrieving episode object for 6x5
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:16] [POSTPROCESS]::SICKBEARD: Snatch history had a quality in it, using that: 1080p HDTV
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: SR snatched this episode and it is not processed before
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: Is ep a priority download: True
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: File /media/SAMSUNG/media/series/Mike & Molly/Season 6/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION.mkv is the same size as /media/SAMSUNG/media/downloaded/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated.mkv
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: This download is marked a priority download so I'm going to replace an existing file if I find one
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: Found the following associated files for /media/SAMSUNG/media/series/Mike & Molly/Season 6/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION.mkv: [u'/media/SAMSUNG/media/series/Mike & Molly/Season 6/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION.pt-BR.srt']
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: Deleting file /media/SAMSUNG/media/series/Mike & Molly/Season 6/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION.mkv
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: Deleting file /media/SAMSUNG/media/series/Mike & Molly/Season 6/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION.pt-BR.srt
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: Found release name Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: Destination folder for this episode: /media/SAMSUNG/media/series/Mike & Molly/Season 6
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: No associated files for /media/SAMSUNG/media/downloaded/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated.mkv were found during this pass
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: Moving file from /media/SAMSUNG/media/downloaded/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated.mkv to /media/SAMSUNG/media/series/Mike & Molly/Season 6/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION.mkv
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: Processing succeeded for /media/SAMSUNG/media/downloaded/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated.mkv
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: Deleted folder: /media/SAMSUNG/media/downloaded/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [POSTPROCESS]::SICKBEARD: Successfully processed
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [INFO]::CLEANDIR: Directory /media/SAMSUNG/media/downloaded/Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated has been processed and removed ...
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: [23:00:17] [INFO]::MAIN: The /home/osmc/SickRage/contrib/nzbToMedia/nzbToMedia.py script completed successfully.
Wed Feb 10 23:00:17 2016	INFO	nzbToMedia: SickBeard: Successfully post-processed Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION.mkv!
Wed Feb 10 23:00:17 2016	INFO	Post-process-script nzbToMedia.py for Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated successful
Wed Feb 10 23:00:17 2016	INFO	Cleaning up download queue for Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated
Wed Feb 10 23:00:17 2016	DETAIL	Deleting file Ih6VZC3xIfFZ8swnggXpGFvcgyw4MwCxRHKJ.vol000+01.par2 from download queue
Wed Feb 10 23:00:17 2016	DETAIL	Deleting file Ih6VZC3xIfFZ8swnggXpGFvcgyw4MwCxRHKJ.vol007+08.par2 from download queue
Wed Feb 10 23:00:17 2016	DETAIL	Deleting file Ih6VZC3xIfFZ8swnggXpGFvcgyw4MwCxRHKJ.vol015+16.par2 from download queue
Wed Feb 10 23:00:17 2016	DETAIL	Deleting file Ih6VZC3xIfFZ8swnggXpGFvcgyw4MwCxRHKJ.vol031+32.par2 from download queue
Wed Feb 10 23:00:17 2016	DETAIL	Deleting file Ih6VZC3xIfFZ8swnggXpGFvcgyw4MwCxRHKJ.vol063+64.par2 from download queue
Wed Feb 10 23:00:17 2016	DETAIL	Deleting file Ih6VZC3xIfFZ8swnggXpGFvcgyw4MwCxRHKJ.vol127+69.par2 from download queue
Wed Feb 10 23:00:17 2016	INFO	Collection Mike.and.Molly.S06E05.1080p.HDTV.X264-DIMENSION-Obfuscated added to history
```

MGaetan89 commented:
@fernandog I'm fine with the API change, as it is an optional new parameters.

fernandog commented:
@miigotu can you review please?

fernandog commented:
@felipesalomao

fernandog commented:
Fixed signature

labrys commented:
If you modify a methods signature or add a new method you are responsible for making sure the docstring is up to date and has all params, *args, and **kwargs documented.

labrys commented:
Approved!
